code/game/objects/items/stunbaton.dm

```diff
@@ -241,9 +241,18 @@
 
 	if(ishuman(L))
 		var/mob/living/carbon/human/H = L
+		var/datum/mind/M = H.mind
+		if(M?.assigned_role == "Assistant" && user.a_intent == INTENT_HARM)
+			var/amount_given = 1
+			if(M.assigned_role == "Clown")
```

Comment:
```suggestion
			if(M?.assigned_role == "Clown")
```

Reply:
we already check this in the previous if statement